core/drand_beacon.go

```diff
@@ -114,36 +114,33 @@ func (bp *BeaconProcess) Load() (bool, error) {
 		return false, err
 	}
 
-	if bp.group != nil {
-		bp.state.Lock()
-		info := chain.NewChainInfo(bp.group)
-		bp.chainHash = info.Hash()
-		bp.state.Unlock()
+	beaconID := bp.getBeaconID()
+	if bp.group == nil {
+		bp.dkgDone = false
+		metrics.DKGStateChange(metrics.DKGNotStarted, beaconID, false)
+		return false, nil
 	}
 
+	bp.state.Lock()
+	info := chain.NewChainInfo(bp.group)
+	bp.chainHash = info.Hash()
 	checkGroup(bp.log, bp.group)
+	bp.state.Unlock()
+
 	bp.share, err = bp.store.LoadShare()
 	if err != nil {
 		return false, err
 	}
 
-	bp.index = int(bp.group.Find(bp.priv.Public).Index)
+	thisBeacon := bp.group.Find(bp.priv.Public)
+	if thisBeacon == nil {
+		return false, fmt.Errorf("could not restore beacon info for the given identity - this can happen if you updated the group file manually")
+	}
+	bp.index = int(thisBeacon.Index)
 	bp.log = bp.log.Named(fmt.Sprint(bp.index))
 
 	bp.log.Debugw("", "serving", bp.priv.Public.Address())
-
-	if bp.group != nil {
-		beaconID := bp.getBeaconID()
-
-		// TODO: can this ever be true?
-		if bp.dkgDone {
-			metrics.DKGStateChange(metrics.DKGDone, beaconID, false)
-		} else {
-			metrics.DKGStateChange(metrics.DKGNotStarted, beaconID, false)
-		}
-	}
-
-	bp.dkgDone = false
+	metrics.DKGStateChange(metrics.DKGDone, beaconID, false)
```

Comment:
That's my understanding - I think it's really a proxy for `shouldHaveAGroupFile`

Reply:
Hmmmm... good question. The only difference that it will make at least ATM is which state is shown in the dashboard. Now that I think about it, if the node doesn't have a group it won't even show up in the dashboard for that beacon, so I think what @AnomalRoil has done is right: when a node starts it should start up as `DKGNotStarted` (meaning it hasn't participated in a DKG since it was started), whereas `DKGDone` is reserved for when the node has finished the DKG at least once.

Reply:
I think that's actually what Patrick has done here, since a node starting for the first time will have `group == nil` and so it will be emitted earlier in that case.